internal/pkg/cli/task_run.go

```diff
@@ -81,6 +87,12 @@ func (o *runTaskOpts) Validate() error {
 		return errMemNotPositive
 	}
 
+	if o.familyName != "" {
+		if err := validateTaskFamilyName(o.familyName); err != nil {
```

Comment:
My thought was that it might be better to provide earlier feedback on naming than later. Since app and svc validate their names during validate and ask, I thought it could also be the case for task.

Since this flag was added after the design meeting (sorryy about the confusion😥), here is a little explanation. The field `familyName` is to task as `AppName` is to app or as `Name` is to svc. It will be later used to name not only task definition, but also the cf stack, log group and the ecr repo. So it needs to be an intersection of the set of valid names for each of these resources.

Reply:
lol it's just a small nit i think but if we decide to keep this validation maybe we should allow underscore? https://github.com/aws/amazon-ecs-cli-v2/pull/1036/files#diff-cf1619746faab25a67237648f6e3594aL141

Reply:
cfn stack name doesn't seem to accept underscore though. If we allow underscore then the name could be potentially invalid for stack?

Reply:
i don't think so. I tried to deploy a CFN template with underscore in my task def name and it worked well. Any reference for the reason why you think it doesn't accept underscore as value (maybe you mean the logicalID doesn't support underscore)? Actually using underscore in family name is pretty common I think :P

Reply:
oh i get you. it's the stack name. Then it makes sense to me!